File: sdks/ruby/lib/nanoservice/errors/error_category.rb
# frozen_string_literal: true

module Nanoservice
  module Errors
    # ErrorCategory classifies the type of error that occurred during node execution.
    module ErrorCategory
      VALIDATION    = "VALIDATION"
      EXECUTION     = "EXECUTION"
      CONFIGURATION = "CONFIGURATION"
      NETWORK       = "NETWORK"
      NOT_FOUND     = "NOT_FOUND"

      ALL = [VALIDATION, EXECUTION, CONFIGURATION, NETWORK, NOT_FOUND].freeze
    end
  end
end

File: sdks/php/src/Errors/ErrorCategory.php
<?php

declare(strict_types=1);

namespace Blok\Nanoservice\Errors;

/**
 * ErrorCategory classifies the type of error that occurred.
 */
enum ErrorCategory: string
{
    case Validation = 'VALIDATION';
    case Execution = 'EXECUTION';
    case Configuration = 'CONFIGURATION';
    case Network = 'NETWORK';
    case NotFound = 'NOT_FOUND';
}
